xbmc/music/MusicDatabase.cpp

```diff
@@ -3380,6 +3383,7 @@ bool CMusicDatabase::GetArtistsByWhere(const std::string& strBaseDir, const Filt
 
         pItem->GetMusicInfoTag()->SetDatabaseId(artist.idArtist, MediaTypeArtist);
         pItem->SetIconImage("DefaultArtist.png");
+        pItem->GetMusicInfoTag()->SetArtist(artist);
```

Comment:
Well it's not working without that line. As I wrote above if there is a better way to do it, please let me know.